master/internal/rm/kubernetesrm/spec.go

```diff
@@ -495,13 +495,27 @@ func (p *pod) createPodSpec(scheduler string) error {
 		WorkingDir:      fluentBaseDir,
 	})
 
+	detContainerSecurityContext := configureSecurityContext(spec.AgentUserGroup)
+	hasContainerSpec := env.RawPodSpec != nil &&
+		env.RawPodSpec.Spec.Containers != nil
```

Comment:
I know that `env.PodSpec()` can be a nil pointer, and you have to check if it's nil, but do you know what types `env.PodSpec().Spec` and `env.PodSpec().Spec.Containers` are?  Are you sure you don't need a nil pointer check on `.Spec`, and you sure you do need the other one?

Reply:
`.Spec` is a struct, so I don't need a nil pointer check.

The length and capacity of a nil slice, or map are 0. So I don't need a nil pointer check on `.Containers`. 
Just for my understanding: The zero value of maps, and slices are nil and in almost all cases cannot be used.
```
var s []int
var m map[int]string
s[3] = 17  // panic
m[5] = 17 // panic
```
s2 := append(s, 2) is a special case. I can append to a nil slice. This is a feature of `append()`.